transport/http/request.go

```diff
@@ -0,0 +1,25 @@
+package http
+
+import (
+	"io"
+	"net/url"
+)
+
+type Header map[string][]string
+
+type Request struct {
+	Method string
+
+	URL  *url.URL
+	Host string
+
+	Header  Header
+	GetBody func() (io.ReadCloser, error)
+}
+
+type Response struct {
+	Request *Request
```

Comment:
Why does the response need to include a pointer to the request?

Reply:
Forgot to clean these up. They are not apart of the middleware design.

Though with that said, this was a pattern the Go HTTP request/response used, and while brainstorming with these transport types i followed that pattern. But probably not needed for a smithy transport library.